### PR TITLE
Add image upload and management endpoints with service layer

### PR DESCRIPTION
**Pull request by LukeHouge**
## Summary
This PR adds comprehensive image upload, retrieval, listing, and deletion functionality to the backend. It includes a new `ImageService` for business logic, REST API endpoints, database models with migrations, Pydantic schemas for request/response validation, and comprehensive unit tests.

### Database & Models
- Added `Image` ORM model to store uploaded images with metadata (filename, MIME type, file size, extracted text, timestamps)
- Added `ImageTag` association table for many-to-many relationship between images and tags
- Created Alembic migration (`a1b2c3d4e5f6`) to create `images` and `image_tags` tables
- Updated `User` and `Tag` models with relationships to images

### Service Layer
- Implemented `ImageService` class with CRUD operations:
  - `upload_image()`: Validates MIME type (JPEG, PNG, GIF, WebP, SVG), enforces 10MB size limit, stores image data
  - `get_image()`: Retrieves single image by ID
  - `list_images()`: Lists images for an owner with pagination
  - `delete_image()`: Deletes image by ID
- Added singleton pattern for service instantiation via `get_image_service()`

### API Routes
- Created `/api/v1/images` endpoints:
  - `POST /api/v1/images`: Upload image with owner_id and optional description
  - `GET /api/v1/images/{image_id}`: Retrieve image metadata
  - `GET /api/v1/images`: List images for owner with pagination
  - `DELETE /api/v1/images/{image_id}`: Delete image
- Proper HTTP status codes (201 for creation, 404 for not found, 400 for validation errors)

### Pydantic Models
- `ImageMetadataResponse`: Full image metadata without binary data
- `ImageUploadResponse`: Response after successful upload
- `ImageListResponse`: Paginated list of images
- `ImageDeleteResponse`: Confirmation of deletion
- All models include field validation (non-negative sizes, etc.)

### Testing
- Unit tests for `ImageService` covering success cases, validation errors, size limits, and edge cases
- Unit tests for image routes covering upload, retrieval, listing, and deletion with mocked dependencies
- Unit tests for Pydantic models validating schema constraints
- Updated database model tests to verify Image and ImageTag tables

### Dependencies
- Added `python-multipart>=0.0.6` for file upload support in FastAPI

### Infrastructure
- Updated test fixtures to clear both OpenAI and image service singletons
- Updated main app to include new image routes

https://claude.ai/code/session_01EE8sc34gXWcC5HUe1peTJe

### CODE REVIEW

File: backend/app/db/models.py [hunk 1]
```diff
@@ -134,6 +138,72 @@ class Tag(Base):
...
+        UUID(as_uuid=True),
+        ForeignKey("tags.id", ondelete="CASCADE"),
+        primary_key=True,
+    )
+
+
+class Image(Base):
+    """Uploaded image model for RAG context."""
+
+    __tablename__ = "images"
```
Comment (LukeHouge) [inline backend/app/db/models.py]: @claude make name more specific, what kind of images?

File: backend/app/models/images.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+from pydantic import BaseModel, Field
+
+
+class ImageMetadataResponse(BaseModel):
+    """Response model for image metadata (without binary data)."""
+
+    id: uuid.UUID = Field(..., description="Image unique identifier")
+    filename: str = Field(..., description="Original filename")
+    description: str | None = Field(None, description="Image description")
```
Comment (LukeHouge) [inline backend/app/models/images.py]: @claude make required with default
